pkg/sanity/resources.go

```diff
@@ -26,19 +27,62 @@ import (
 
 	"github.com/container-storage-interface/spec/lib/go/csi"
 
+	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
-// VolumeInfo keeps track of the information needed to delete a volume.
-type VolumeInfo struct {
+// managedResourceInfos is a map of managed resource infos keyed by resource
+// IDs.
+type resourceInfos map[string]resourceInfo
+
+// getLastInserted returns the last inserted resourceInfo, or nil if
+// resourceInfos is empty.
+func (ris resourceInfos) getLastInserted() *resourceInfo {
+	if len(ris) == 0 {
+		return nil
+	}
+
+	var last *resourceInfo
+	for _, ri := range ris {
+		if last == nil || ri.order > last.order {
+			last = &ri
+		}
+	}
+	return last
+}
+
+// addItem adds a resourceInfo item by the given ID and with the given data.
+func (ris resourceInfos) addItem(id string, data interface{}) {
+	nextOrder := -1
+	if lastInserted := ris.getLastInserted(); lastInserted != nil {
+		nextOrder = lastInserted.order
+	}
+	nextOrder++
+
+	ris[id] = resourceInfo{
+		id:    id,
+		order: nextOrder,
+		data:  data,
+	}
+}
+
+// resourceInfo represents a resource (i.e., a volume or a snapshot).
+type resourceInfo struct {
+	id    string
+	order int
+	data  interface{}
+}
+
+// volumeInfo keeps track of the information needed to delete a volume.
+type volumeInfo struct {
```

Comment:
I unexported the type since it's effectively an implementation detail. I guess strictly speaking, this could be considered a breaking change, though I doubt anyone uses it (or rather hope not so).

If we're concerned about this, I'm also happy to undo the change and safe it for when we have more things to change that justify a new major release.